kernel/socket_kern.c

```diff
@@ -290,20 +290,13 @@ static inline void update_pid_bandwidth(__u64 sent, __u64 received, __u8 protoco
     __u32 key = NETDATA_CONTROLLER_APPS_ENABLED;
 
     __u32 *apps = bpf_map_lookup_elem(&socket_ctrl ,&key);
-    if (apps) {
-        if (*apps == 0)
-            return;
-    } else
+    if (!apps || (apps && *apps == 0))
```

Comment:
```suggestion
    if (!apps || *apps == 0)
```
`apps` is not `NULL` after the first check. No need to check it again.

Reply:
Thank you, I addressed the upgrade with last commit!